src/rust/ide/lib/view/src/graph_editor/component/breadcrumbs.rs

```diff
@@ -0,0 +1,366 @@
+                for index in current_index..index {
+                    let info = self.breadcrumbs.borrow().get(index).map(|breadcrumb| {
+                        let definition = breadcrumb.info.method_pointer.clone();
+                        let call       = breadcrumb.info.expression_id;
+                        LocalCall{call,definition}
+                    }).as_ref().cloned();
+                    if info.is_some() {
+                        self.frp_outputs.breadcrumb_push.emit(info);
+                    } else {
+                        error!(self.logger, "LocalCall info is invalid.");
+                        self.remove_breadcrumbs_history(index);
+                        break;
+                    }
+                }
+            },
+            cmp::Ordering::Equal => ()
+        }
+        info!(self.logger,"Selecting breadcrumb #{index}");
+        sources
+    }
+
+    fn push_breadcrumb(&self, local_call:&Option<LocalCall>) -> Vec<frp::Source> {
+        let mut sources = Vec::new();
+        if let Some(local_call) = local_call {
+            let method_pointer = &local_call.definition;
+            let expression_id  = &local_call.call;
+            let current_index = self.current_index.get();
+            let next_index = current_index + 1;
+
+            let breadcrumb_exists =
+                self.breadcrumbs.borrow_mut().get(current_index).contains_if(|breadcrumb| {
+                    breadcrumb.info.expression_id == *expression_id
+                });
+
+            if breadcrumb_exists {
+                debug!(self.logger, "Entering an existing {method_pointer.name} breadcrumb.");
+                //TODO[dg]: Highlight breadcrumb.
+            } else {
+                debug!(self.logger, "Creating a new {method_pointer.name} breadcrumb.");
+                self.remove_breadcrumbs_history(self.current_index.get());
+                let breadcrumb = Breadcrumb::new(&self.scene, method_pointer, expression_id);
+                let network = &breadcrumb.frp.network;
+                let breadcrumb_index = next_index;
+                let model = self.clone_ref();
+
+
+                // === User Interaction ===
+
+                frp::extend! { network
+                    eval_ breadcrumb.frp.outputs.selected(
+                        model.select_breadcrumb(breadcrumb_index).iter().for_each(|source| {
+                            source.emit(())
+                        })
+                    );
+                }
+
+                info!(self.logger, "Pushing {breadcrumb.info.method_pointer.name} breadcrumb.");
+                breadcrumb.set_position(Vector3(self.width(),0.0,0.0));
+                sources.push(breadcrumb.frp.fade_in.clone_ref());
+                self.breadcrumbs_container.add_child(&breadcrumb);
+                self.breadcrumbs.borrow_mut().push(breadcrumb);
+            }
+            self.current_index.set(next_index);
+            sources.append(&mut self.update_selection());
```

Comment:
Here, the `update_selection` should return 2 vectors - one containing IDs of breadcrumbs to be deselected, and the second to be selected.